modules/HardwareDrivers/PowerSupplies/Huawei_V100R023C10/fusion_charger_lib/modbus-server/libs/server/CMakeLists.txt

```diff
@@ -0,0 +1,18 @@
+file(GLOB_RECURSE MODBUS_SERVER_SOURCES "src/*.cpp")
+
+add_library(
+    modbus-server
+    STATIC
+        ${MODBUS_SERVER_SOURCES}
+)
+target_include_directories(modbus-server PUBLIC include)
+target_link_libraries(modbus-server PUBLIC modbus-base Huawei::FusionCharger::LogInterface)
+
+if(BUILD_TESTING)
+    if(TEST_COVERAGE)
+        # Add coverage flags
+        target_compile_options(modbus-server PUBLIC --coverage)
+        target_link_options(modbus-server PUBLIC --coverage)
+    endif()
+    add_subdirectory(tests)
+endif()
```

Comment:
see comments about testing and coverage, https://github.com/EVerest/everest-core/pull/1336#discussion_r2309729278 still applies